### PR TITLE
expansion and update to transformer tutorial

### PR DESCRIPTION
**Pull request by nslatysheva**
Some quick elaborations/fixes on the old tutorial version. Main changes: 
+ Automating the data downloads, unzips, and file and directory shuffling by adding these lines into the bash script 
+ Copying over the instructions for building SentencePiece from source
+ Describing how to start Tensorboard monitoring (+ a link to a tutorial for users unfamiliar with it)
+ Added command for averaging parameters of several checkpointed models, required for the inference script
+ Mentioning that one should change the GPU number within `run_wmt_ende.sh` to match your config

### CODE REVIEW

File: scripts/wmt/README.md [hunk 1]
```diff
@@ -1,49 +1,115 @@
...
+
+If you are using a Docker container, you can launch a Tensorboard run from outside the container using:
+
+`docker exec <CONTAINER ID> tensorboard --logdir="path/to/wmt_ende_transformer" --port=6006`
+
+Note that for this to work, you need to have exposed the port when you create your Docker container ([simple example](https://briancaffey.github.io/2017/11/20/using-tensorflow-and-tensor-board-with-docker.html)).
+
+### Translation using a trained model 
+
+You can run the following script to perform inference on the test set using a trained model. The script calls `onmt-main infer`. Normally, the latest checkpoint is used for inference by default, but [we recommend averaging the parameters of several checkpoints](http://opennmt.net/OpenNMT-tf/inference.html#checkpoints-averaging), which usually boosts model performance.
```
Comment (guillaumekln) [inline scripts/wmt/README.md]: You should mention that the training does that automatically if run until the end.

File: scripts/wmt/prepare_data.sh [hunk 2]
```diff
@@ -19,20 +21,54 @@
...
+# set vocabulary size and source and target languages
 vocab_size=32000
 sl=en
 tl=de
 
+# Download the default datasets into the $DATA_PATH; mkdir if it doesn't exist
+mkdir -p $DATA_PATH
+cd $DATA_PATH
+#
+# echo "Downloading and extracting Commoncrawl data (919 MB) for training..."
```
Comment (guillaumekln) [inline scripts/wmt/prepare_data.sh]: I think you should uncomment this download as it is uncommented in the `corpus` set.

File: scripts/wmt/prepare_data.sh [hunk 3]
```diff
@@ -19,20 +21,54 @@
...
+# wget --trust-server-names http://www.statmt.org/wmt13/training-parallel-commoncrawl.tgz
+# tar zxvf training-parallel-commoncrawl.tgz
+# ls | grep -v 'commoncrawl.de-en.[de,en]' | xargs rm
+#
+# echo "Downloading and extracting Europarl data (658 MB) for training..."
```
Comment (guillaumekln) [inline scripts/wmt/prepare_data.sh]: Same here.

File: scripts/wmt/README.md [hunk 4]
```diff
@@ -1,49 +1,115 @@
...
 
 ```bash
 CUDA_VISIBLE_DEVICES=0 ./run_wmt_ende_1gpu.sh
 ```
 
-### Translate
+The training logs printed to STDOUT include the values:
++ `words_per_sec/features`: number of input words processed per second (where features = words in the training set)
++ `words_per_sec/labels`: number of output words processed per second (where labels = correct words in the target language train file)
++ `global_step/sec`: number of gradient updates/# batches processed per second. `global step` is a Tensorflow variable that counts the number of batches seen by the model.
```
Comment (guillaumekln) [inline scripts/wmt/README.md]: Change to "number of parameter updates. `global step` is a TensorFlow variable that is incremented each time the gradients are applied.".

File: scripts/wmt/README.md [hunk 5]
```diff
@@ -1,49 +1,115 @@
...
+
 Or if you have only 1 GPU, run the dedicated script:
 
 ```bash
 CUDA_VISIBLE_DEVICES=0 ./run_wmt_ende_1gpu.sh
 ```
 
-### Translate
+The training logs printed to STDOUT include the values:
++ `words_per_sec/features`: number of input words processed per second (where features = words in the training set)
```
Comment (guillaumekln) [inline scripts/wmt/README.md]: Change to "number of source tokens processed per second" and remove the content in parenthesis ("source" is clear enough).

File: scripts/wmt/README.md [hunk 6]
```diff
@@ -1,49 +1,115 @@
...
++ `words_per_sec/labels`: number of output words processed per second (where labels = correct words in the target language train file)
```
Comment (guillaumekln) [inline scripts/wmt/README.md]: Change to "number of target tokens processed per second" and remove the content in parenthesis ("target" is clear enough).

File: scripts/wmt/prepare_data.sh (hunk 2 above)
Comment (nslatysheva) [reply scripts/wmt/prepare_data.sh]: Yes, forgot about uncommenting these lines, thanks for spotting!

**Comment by guillaumekln** [comment]
Thanks!
